Zilon.Core/Zilon.Core/MapGenerators/StaticObstaclesGenerator.cs

```diff
@@ -69,5 +53,46 @@ public Task CreateAsync(ISector sector, ISectorSubScheme sectorSubScheme)
 
             return Task.CompletedTask;
         }
+
+        private IStaticObject CreateStaticObject(ISector sector, HexNode node)
+        {
+            var staticObjectPurpose = RollPurpose();
+
+            var factory = SelectStaticObjectFactory(staticObjectPurpose);
+
+            var staticObject = factory.Create(sector, node, default);
+
+            return staticObject;
+        }
+
+        private IStaticObjectFactory SelectStaticObjectFactory(PropContainerPurpose staticObjectPurpose)
+        {
+            var factories = _staticObjectfactoryCollector.GetFactories();
+
+            foreach (var factory in factories)
+            {
+                if (factory.Purpose != staticObjectPurpose)
+                {
+                    continue;
+                }
+
+                return factory;
+            }
+
+            throw new InvalidOperationException($"Не обнаружена фабрика для статических объектов типа {staticObjectPurpose}");
+        }
+
+        private PropContainerPurpose RollPurpose()
+        {
+            var availableStatiObjectPurpose = new[] {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'availableStatiObjectPurpose'.](https://app.codacy.com/manual/kreghek/Zilon_Roguelike/pullRequest?prid=5384689)